### PR TITLE
Prevent user from deleting pre-defined networks

### LINKED ISSUE #17255: Able to remove the default bridge network
* I was able to remove the default bridge network, now i can't run continers
* Trying to remove `null` says it can't be found but it clearly is found by `docker network ls`

```
buntu@ip-172-31-36-118:~$ docker network ls
NETWORK ID          NAME                DRIVER
a792d9893ebd        none                null                
37c1b62f7b40        host                host                
97ee5a4064f7        bridge              bridge              
ubuntu@ip-172-31-36-118:~$ docker network rm bridge
ubuntu@ip-172-31-36-118:~$ docker network ls
NETWORK ID          NAME                DRIVER
a792d9893ebd        none                null                
37c1b62f7b40        host                host                
ubuntu@ip-172-31-36-118:~$ docker network rm host
Error response from daemon: network of type "host" cannot be deleted
ubuntu@ip-172-31-36-118:~$ docker network rm null
Error response from daemon: network null not found
ubuntu@ip-172-31-36-118:~$ docker network ls
NETWORK ID          NAME                DRIVER
a792d9893ebd        none                null                
37c1b62f7b40        host                host                
ubuntu@ip-172-31-36-118:~$ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
b901d36b6f2f: Pull complete 
0a6ba66e537a: Pull complete 
Digest: sha256:517f03be3f8169d84711c9ffb2b3235a4d27c1eb4ad147f6248c8040adb93113
Status: Downloaded newer image for hello-world:latest
Error response from daemon: Cannot start container da483ad019745e76834d1b2bd4be6b742367b5b53d985f306d5c596fa08a180b: network bridge not found
ubuntu@ip-172-31-36-118:~$ 

```

### PR DESCRIPTION
**Pull request by mavenugo**
fixes #17255 
Signed-off-by: Madhu Venugopal madhu@docker.com

### CODE REVIEW

File: integration-cli/docker_api_network_test.go [hunk 1]
```diff
@@ -193,6 +193,23 @@ func (s *DockerSuite) TestApiNetworkIpamMultipleBridgeNetworks(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestApiCreateDeletePredefinedNetworks(c *check.C) {
+	ceateDeletePredefinedNetwork(c, "bridge")
+	ceateDeletePredefinedNetwork(c, "none")
+	ceateDeletePredefinedNetwork(c, "host")
+}
+
+func ceateDeletePredefinedNetwork(c *check.C, name string) {
```
Comment (thaJeztah) [inline integration-cli/docker_api_network_test.go]: Typo; s/ceate/create/

Comment (thaJeztah) [inline integration-cli/docker_api_network_test.go]: I think? :smile:

Comment (mavenugo) [inline integration-cli/docker_api_network_test.go]: :) yes. will fix it.

**Comment by tiborvass** [comment]
LGTM

**Comment by LK4D4** [comment]
Honestly, daemon methods should check it and return error. I don't think that importing more things to API will make any good.
However we can fix it later if you want.

**Comment by mavenugo** [comment]
@LK4D4 yes, we could. but, some of the APIs call the libnetwork library directly for straight-forward operations without a daemon wrapper to it. We could do that at a later point if that is okay.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**